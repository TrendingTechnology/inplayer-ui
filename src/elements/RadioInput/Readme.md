RadioInput is a input component for radio. You can use it directly like this:

```jsx static
import { RadioInput } from 'inplayer-ui';

function Page(props) {
  return <RadioInput>Buy now</RadioInput>;
}
```

### Demo

```jsx
<RadioInput>Buy now</RadioInput>
```
