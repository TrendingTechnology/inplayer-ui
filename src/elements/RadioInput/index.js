import styled from 'styled-components';
import { uiColors } from 'utils';
import { applyStyleModifiers } from 'styled-components-modifiers';

const modifiers = {
  hoverDisabled: () => `
    &:hover {
      background: ${uiColors('lightGray')};
      border: 1px solid ${uiColors('gray')};
      pointer-events: none;
    }
  `,
};

const RadioInput = styled.input`
  display: flex;
  background: ${uiColors('white')};
  font-weight: 300;
  border: 1px solid ${uiColors('gray')};
  width: 16px;
  height: 16px;
  border-radius: 14px;
  cursor: pointer;
  position: relative;
  transition: all ease 300ms;

  &::after {
    content: '';
    display: block;
    position: absolute;
    background: ${uiColors('white')};
    transform: rotate(45deg);
    top: 2px;
    left: 2px;
    width: 10px;
    height: 10px;
    border-radius: 14px;
    transition: all ease 300ms;
  }

  &:checked {
    border: 1px solid ${uiColors('accentBlue')};
  }

  &:checked::after {
    background: ${uiColors('accentBlue')};
  }

  &:hover {
    background: ${uiColors('accentBlueFade')};
    border: 1px solid ${uiColors('accentBlue')};
  }

  ${applyStyleModifiers(modifiers)};
`;

/** @component */
export default RadioInput;
