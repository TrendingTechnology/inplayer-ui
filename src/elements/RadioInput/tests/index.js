import { generateSnapshots } from 'helpers';

import RadioInput from '../index';

generateSnapshots(RadioInput);
