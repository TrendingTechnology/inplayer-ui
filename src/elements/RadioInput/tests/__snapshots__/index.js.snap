// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioInput renders correctly 1`] = `
.c0 {
  display: inline-block;
  background: #fff;
  font-weight: 300;
  border: 1px solid #d7dde5;
  width: 16px;
  height: 16px;
  border-radius: 14px;
  cursor: pointer;
  position: relative;
  -webkit-transition: all ease 300ms;
  transition: all ease 300ms;
}

.c0::after {
  content: '';
  display: block;
  position: absolute;
  background: #fff;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  top: 2px;
  left: 2px;
  width: 10px;
  height: 10px;
  border-radius: 14px;
  -webkit-transition: all ease 300ms;
  transition: all ease 300ms;
}

.c0:checked {
  border: 1px solid #00aae6;
}

.c0:checked::after {
  background: #00aae6;
}

.c0:hover {
  background: #00aae60f;
  border: 1px solid #00aae6;
}

<input
  className="c0"
/>
`;
